emscripten.py

```diff
@@ -441,16 +435,21 @@ def emscript(infile, outfile_js, memfile, temp_files, DEBUG):
   # merge forwarded data
   shared.Settings.EXPORTED_FUNCTIONS = forwarded_json['EXPORTED_FUNCTIONS']
 
-  exports = metadata['exports']
+  function_exports = metadata['exports']
+
+  # The table export is not listed in the 'exports' but we still want to avoid DCE'ing it in the
+  # same way we do other exports.
```

Comment:
I don't think the line below affects DCE? it just is necessary for the JS to receive the export. DCE later may still remove it in theory (but that would be fine if it did)

Reply:
I need to go back and look at why I need this change, but the place its is really needed is in building.py:metadce when extra_info is constructed:

```
extra_info = '{ "exports": [' + ','.join(map(lambda x: '["' + x[0] + '","' + x[1] + '"]', Settings.MODULE_EXPORTS)) +      ']}'   
```

Without this we either don't remove the export, or we don't minify or something.. I need to go back and check.

Reply:
reverted this part.